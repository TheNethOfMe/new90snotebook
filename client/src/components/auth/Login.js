import React, { Component } from "react";

class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: "",
      password: "",
      errors: {}
    };
  }
  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };
  onSubmit = e => {
    e.preventDefault();
    const loggedUser = {
      email: this.state.email,
      password: this.state.password
    };
    console.log(loggedUser);
  };
  render() {
    return (
      <div>
        <h1>Login</h1>
        <form onSubmit={this.onSubmit}>
          <input
            type="text"
            name="email"
            value={this.state.email}
            onChange={this.onChange}
          />
          <input
            type="password"
            name="password"
            value={this.state.password}
            onChange={this.onChange}
          />
          <input type="submit" value="Log In" />
        </form>
      </div>
    );
  }
}

export default Login;
