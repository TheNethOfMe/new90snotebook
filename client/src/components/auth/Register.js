import React, { Component } from "react";
import axios from "axios";
import classnames from "classnames";

class Register extends Component {
  constructor() {
    super();
    this.state = {
      email: "",
      password: "",
      password2: "",
      errors: {}
    };
  }
  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };
  onSubmit = e => {
    e.preventDefault();
    const newUser = {
      email: this.state.email,
      password: this.state.password,
      password2: this.state.password2
    };
    axios
      .post("/api/users/register", newUser)
      .then(res => console.log(res.data))
      .catch(err => {
        this.setState({ errors: err.response.data });
      });
  };
  render() {
    const { errors } = this.state;
    return (
      <div>
        <h1>Register</h1>
        <form noValidate onSubmit={this.onSubmit}>
          <input
            type="email"
            className={classnames({ "invalid-input": errors.email })}
            name="email"
            value={this.state.email}
            onChange={this.onChange}
          />
          {errors.email && <div>{errors.email}</div>}
          <input
            type="password"
            className={classnames({ "invalid-input": errors.password })}
            name="password"
            value={this.state.password}
            onChange={this.onChange}
          />
          {errors.password && <div>{errors.password}</div>}
          <input
            type="password"
            className={classnames({ "invalid-input": errors.password2 })}
            name="password2"
            value={this.state.password2}
            onChange={this.onChange}
          />
          {errors.password2 && <div>{errors.password2}</div>}
          <input type="submit" value="Sign Up" />
        </form>
      </div>
    );
  }
}

export default Register;
