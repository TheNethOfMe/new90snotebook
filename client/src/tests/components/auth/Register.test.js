import React from "react";
import { shallow } from "enzyme";
import { Register } from "../../../components/auth/Register";
import { fakeNewUser } from "../../dummyData";

let wrapper, registerUser, auth, errors, history;

beforeEach(() => {
  registerUser = jest.fn();
  auth = { user: "test", isAuthenticated: false };
  errors = {};
  history = {};
  wrapper = shallow(
    <Register
      registerUser={registerUser}
      auth={auth}
      errors={errors}
      history={history}
    />
  );
});

test("should render the Register page", () => {
  expect(wrapper).toMatchSnapshot();
});

test("should write text to inputs", () => {
  const testVal = "TestingValue";
  expect(wrapper.state().email).toBeFalsy();
  expect(wrapper.state().password).toBeFalsy();
  expect(wrapper.state().password2).toBeFalsy();
  wrapper
    .find("input")
    .at(0)
    .simulate("change", { target: { name: "email", value: testVal } });
  expect(wrapper.state().email).toEqual(testVal);
  wrapper
    .find("input")
    .at(1)
    .simulate("change", { target: { name: "password", value: testVal } });
  expect(wrapper.state().password).toEqual(testVal);
  wrapper
    .find("input")
    .at(2)
    .simulate("change", { target: { name: "password2", value: testVal } });
  expect(wrapper.state().password2).toEqual(testVal);
});

test("should call loginUser on form submit", () => {
  wrapper.setState(fakeNewUser);
  wrapper.find("form").simulate("submit", { preventDefault: () => jest.fn() });
  expect(registerUser).toHaveBeenCalledWith(fakeNewUser, history);
});
