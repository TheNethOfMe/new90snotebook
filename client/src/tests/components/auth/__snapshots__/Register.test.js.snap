// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Register page 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Register
    auth={
      Object {
        "isAuthenticated": false,
        "user": "test",
      }
    }
    errors={Object {}}
    history={Object {}}
    registerUser={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Register
        </h1>,
        <form
          noValidate={true}
          onSubmit={[Function]}
        >
          <input
            className=""
            name="email"
            onChange={[Function]}
            type="email"
            value=""
          />
          <input
            className=""
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
          <input
            className=""
            name="password2"
            onChange={[Function]}
            type="password"
            value=""
          />
          <input
            type="submit"
            value="Sign Up"
          />
        </form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Register",
        },
        "ref": null,
        "rendered": "Register",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className=""
              name="email"
              onChange={[Function]}
              type="email"
              value=""
            />,
            undefined,
            <input
              className=""
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />,
            undefined,
            <input
              className=""
              name="password2"
              onChange={[Function]}
              type="password"
              value=""
            />,
            undefined,
            <input
              type="submit"
              value="Sign Up"
            />,
          ],
          "noValidate": true,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "",
              "name": "email",
              "onChange": [Function],
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "",
              "name": "password",
              "onChange": [Function],
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "",
              "name": "password2",
              "onChange": [Function],
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "type": "submit",
              "value": "Sign Up",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Register
          </h1>,
          <form
            noValidate={true}
            onSubmit={[Function]}
          >
            <input
              className=""
              name="email"
              onChange={[Function]}
              type="email"
              value=""
            />
            <input
              className=""
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />
            <input
              className=""
              name="password2"
              onChange={[Function]}
              type="password"
              value=""
            />
            <input
              type="submit"
              value="Sign Up"
            />
          </form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Register",
          },
          "ref": null,
          "rendered": "Register",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className=""
                name="email"
                onChange={[Function]}
                type="email"
                value=""
              />,
              undefined,
              <input
                className=""
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />,
              undefined,
              <input
                className=""
                name="password2"
                onChange={[Function]}
                type="password"
                value=""
              />,
              undefined,
              <input
                type="submit"
                value="Sign Up"
              />,
            ],
            "noValidate": true,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "",
                "name": "email",
                "onChange": [Function],
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "",
                "name": "password",
                "onChange": [Function],
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "",
                "name": "password2",
                "onChange": [Function],
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "type": "submit",
                "value": "Sign Up",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
